src/java.base/share/classes/jdk/internal/crac/JDKContext.java

```diff
@@ -35,7 +35,7 @@
     static class ContextComparator implements Comparator<Map.Entry<JDKResource, Void>> {
         @Override
         public int compare(Map.Entry<JDKResource, Void> o1, Map.Entry<JDKResource, Void> o2) {
-            return o2.getKey().getPriority() - o1.getKey().getPriority();
+            return o1.getKey().getPriority().ordinal() - o2.getKey().getPriority().ordinal();
```

Comment:
Enums are already comparable so we can use the built in compareTo method here but this isn't critical and shouldn't block merging

```suggestion
            return o1.getKey().getPriority().compareTo(o2.getKey().getPriority());
```

Reply:
Hi @DanHeidinga !
Thank you for review. Yes, compareTo looks better. Updated.